dace/subsets.py

```diff
@@ -895,6 +909,10 @@ def size(self):
     def size_exact(self):
         return self.size()
 
+    def volume_exact(self) -> int:
+        """ Returns the total number of elements in all dimenssions together. """
+        return reduce(operator.mul, self.size_exact())
```

Comment:
Is this compatible with all supported Python versions? Note that we also have another method that does the same in `dace.data`: https://github.com/spcl/dace/blob/057a6804ea2da60b053895e490cf230d0ef90225/dace/data.py#L139
Maybe it is a good idea to move (one of them) to a utility method.

Reply:
As far as I can tell, it does work for 3.7 and 3.12. Replaced with lambda function regardless, since that's how `_prod` works.

I don't think it's necessary to make `reduce(lambda a, b : a * b, sequence)` itself an utility function. It's really short, easy to see what it does. I've added the `volume_exact()` function to just give that operation a meaningful name (i.e. why it does that). But of course, if you rather have it separately, I can move it too.